Constructor.cc

```diff
@@ -828,74 +686,68 @@ Constructor::is_linear()
 void
 Constructor::set_in_selection(bool state)
 {
-    for (Vars_iter i = d_vars.begin(); i != d_vars.end(); i++) {
-        (*i)->set_in_selection(state);
+    for (auto var: d_vars) {
+        var->set_in_selection(state);
     }
 
     BaseType::set_in_selection(state);
 }
 
-
 void Constructor::transfer_attributes(AttrTable *at_container)
 {
     AttrTable *at = at_container->get_attr_table(name());
-    DBG(cerr << "Constructor::transfer_attributes() - processing " << name() << "'  addr: "<< (void*) at << endl);
+
     if (at) {
         BaseType::transfer_attributes(at_container);
-        for (Vars_iter i = d_vars.begin(); i != d_vars.end(); i++) {
-            BaseType *bt  = (*i);
-            bt->transfer_attributes(at);
+        for (auto var: d_vars) {
+            var->transfer_attributes(at);
         }
-
     }
 }
 
 AttrTable *
-Constructor::make_dropped_vars_attr_table(vector<BaseType *> *dropped_vars) {
-    DBG( cerr << __func__ << "() - BEGIN" << endl;);
-
-    AttrTable *dv_table = NULL;
-    if(!dropped_vars->empty()){
+Constructor::make_dropped_vars_attr_table(vector<BaseType *> *dropped_vars)
+{
+    AttrTable *dv_table = nullptr;
+    if (!dropped_vars->empty()) {
         dv_table = new AttrTable;
         dv_table->set_name("dap4:dropped_members");
+
         vector<BaseType *>::iterator dvIter = dropped_vars->begin();
         vector<BaseType *>::iterator dvEnd = dropped_vars->end();
         unsigned int i = 0;
-        for( ; dvIter!=dvEnd ; dvIter++, i++){
+        for (; dvIter != dvEnd; dvIter++, i++) {
             BaseType *bt = (*dvIter);
+
             AttrTable *bt_attr_table = new AttrTable(bt->get_attr_table());
             bt_attr_table->set_name(bt->name());
             string type_name = bt->type_name();
-            if(bt->is_vector_type()){
+
+            if (bt->is_vector_type()) {
                 Array *array = dynamic_cast <Array *>(bt);
-                if(array){
+                if (array) {    // This is always true - only an Array is_vector_type(). jhrg 4/25/22
```

Comment:
If it's always true then why continue testing it?

Reply:
I was hesitant to change that code in this PR.